visualization/app/codeCharta/ui/nodeContextMenu/nodeContextMenu.component.ts

```diff
@@ -62,82 +59,96 @@ export class NodeContextMenuController
 		CodeMapMouseEventService.subscribeToBuildingRightClickedEvents(this.$rootScope, this)
 		NodeContextMenuController.subscribeToShowNodeContextMenu(this.$rootScope, this)
 		NodeContextMenuController.subscribeToHideNodeContextMenu(this.$rootScope, this)
-		BlacklistService.subscribe($rootScope, this)
-		FocusedNodePathService.subscribeToFocusNode($rootScope, this)
-		FocusedNodePathService.subscribeToUnfocusNode($rootScope, this)
-		MarkedPackagesService.subscribe($rootScope, this)
-
-		document.body.addEventListener(
-			"click",
-			event => NodeContextMenuController.broadcastHideEvent(this.$rootScope, new Vector2(event.x, event.y)),
-			true
-		)
 	}
 
 	onMapColorsChanged(mapColors: MapColors) {
 		this._viewModel.markingColors = mapColors.markingColors
 	}
 
-	onBlacklistChanged() {
-		this.onHideNodeContextMenu()
-	}
-
-	onFocusNode() {
-		this.onHideNodeContextMenu()
-	}
-
-	onUnfocusNode() {
-		this.onHideNodeContextMenu()
-	}
-
-	onMarkedPackagesChanged() {
-		this.onHideNodeContextMenu()
-	}
-
 	onBuildingRightClicked(building: CodeMapBuilding, x: number, y: number) {
 		const nodeType = building.node.isLeaf ? NodeType.FILE : NodeType.FOLDER
 		this.onShowNodeContextMenu(building.node.path, nodeType, x, y)
 	}
 
 	onShowNodeContextMenu(path: string, nodeType: string, mouseX: number, mouseY: number) {
-		NodeContextMenuController.broadcastHideEvent(this.$rootScope)
 		this._viewModel.codeMapNode = getCodeMapNodeFromPath(path, nodeType, this.codeMapPreRenderService.getRenderMap())
+		this._viewModel.showNodeContextMenu = true
+
 		const { x, y } = this.calculatePosition(mouseX, mouseY)
 		this.setPosition(x, y)
+
 		this.synchronizeAngularTwoWayBinding()
+
+		// Add event listeners, so that opened node context menu can be closed again later
+		// when clicking (left or right button)
+		// or using the mouse wheel on the body element.
+		document.body.addEventListener("click", this.onBodyLeftClickHideNodeContextMenu, true)
+		document.body.addEventListener("mousedown", this.onBodyRightClickHideNodeContextMenu, true)
+		document.body.addEventListener("wheel", this.onBodyWheelHideNodeContextMenu, true)
```

Comment:
Would it not be sufficient to add the event listener not to the body but to the dom element that we want to track the wheel upon?

Reply:
I checked this. For the wheel listener, I agree.
The other listeners should be added to the body element, so that clicks on the ribbon bar and other interfaces will close the node context menu.